node/src/components/block_proposer.rs

```diff
@@ -140,10 +145,16 @@ where
         match (&mut self.state, event) {
             (
                 BlockProposerState::Initializing { ref mut pending },
-                Event::Loaded { chainspec, sets },
+                Event::Loaded {
+                    chainspec,
+                    sets,
+                    next_finalized_block,
+                },
             ) => {
                 let mut new_ready_state = BlockProposerReady {
-                    sets: sets.unwrap_or_default(),
+                    sets: sets
+                        .unwrap_or_default()
+                        .with_next_finalized(next_finalized_block),
```

Comment:
So we persist that but then always override it on startup? :confused:

Reply:
Yeah. A bit suboptimal, we probably shouldn't persist it - would make more sense to take it out of the state that gets persisted.